### PR TITLE
Custom embedding API

### PR DESCRIPTION
**Pull request by drio18**
Creates new API to selectively embed fields from an item's @@object view, enabling quicker calls to the database for items with costly @@embedded views. The main parameters to be supplied by the POST are:
- `id`: a uuid if POSTing with url parameters; a list of either uuids or @ids if POST with body parameters
- `depth`: the maximum depth of the embedding, with a depth of 0 returning the @@object view of the item and depth increasing by 1 when embedding a new item within a chain (so variant sample -> variant -> gene has a depth of 2 from variant sample to gene)
- `desired`: specifies the @ids to embed and should be formatted according to @id convention, so if embedding a variant inside a variant sample "variants" would be the correct format
- `ignored`: specifies the @ids to not embed, formatted similar to `desired`; will only be used if `desired` is not supplied.

If parameters such as `depth` or `desired` are not supplied, embedding will proceed according to default settings, but these parameters are not guaranteed to work and also will likely not be as fast as providing only the parameters required for the purpose of the call to the API. 

Additionally, this PR changes the implementation of the variant sample calculated property `project_genelists` such that calls to the database for variant sample lists will no longer be embedding gene lists. This solves the performance issues for variant sample lists, but variant samples/variants/genes still have gene lists embedded and thus will have long response times if large gene lists are included.

### CODE REVIEW

File: src/encoded/quick_embed.py [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+    embed_depth = 4  # Arbritary standard depth to search.
+    ignored_keys = KEYS_TO_IGNORE
+    ignored(context)
+    if request.GET:
+        ids += request.GET.dict_of_lists().get("id", [])
+        embed_depth = int(request.GET.get("depth", embed_depth))
+        ignored_embeds += request.GET.dict_of_lists().get("ignored", [])
+        desired_embeds += request.GET.dict_of_lists().get("desired", [])
+    elif request.json:
+        ids += request.json.get("ids", [])
```
Comment (willronchetti) [inline src/encoded/quick_embed.py:157]: Let's limit the # of ID's that can be passed (and ensure this limitation is documented) - if more are passed feel free to throw HTTPBadRequest. I think 5 is a reasonable number? Are there any APIs we can envision wanting to request more than 5 at once?

File: src/encoded/tests/test_quick_embed.py [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+
+from encoded.quick_embed import ATID_PATTERN, MINIMAL_EMBEDS
+
+pytestmark = [pytest.mark.working]
+
+EMBED_URL = "/embed"
+KEYS_NOT_INCLUDED = ["@context", "actions", "aggregated-items", "validation-errors"]
+
+
+def _embed_with_url_params(testapp, embed_string, status="*"):
```
Comment (willronchetti) [inline src/encoded/tests/test_quick_embed.py:11]: A class for the test file would be advantageous as well.

File: src/encoded/tests/test_quick_embed.py [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+
+def test_depth_of_zero(testapp, variant_sample_list):
+    """Test request with depth=0 returns @@object view of item."""
+    vsl_uuid = variant_sample_list["uuid"]
+    url_params = EMBED_URL + "?id=" + vsl_uuid + "&depth=0"
+    vsl_embed = _embed_with_url_params(testapp, url_params)
+    for key, value in variant_sample_list.items():
+        if key not in KEYS_NOT_INCLUDED:
+            assert value == vsl_embed[key]
+
```
Comment (willronchetti) [inline src/encoded/tests/test_quick_embed.py:103]: Might consider adding a test for multiple uuids.

File: src/encoded/types/variant.py [hunk 4]
```diff
@@ -498,17 +498,36 @@ def extract_chrom_from_variant(v):
         }
     })
     def project_genelists(self, request, project, variant):
+        gene_atids = []
```
Comment (willronchetti) [inline src/encoded/types/variant.py:501]: Probably a good idea to add a docstring.

**Review by willronchetti — APPROVED** [review]
Approving with non-blocking comments

**Review by Bianca-Morris — COMMENTED** [review]
Code seems reasonable; I haven't gotten a chance to test this yet, and probably won't be able to for a couple more days yet. I think if you and Will are happy with it, we can go ahead and merge it, and then if I have any questions/suggestions, we can add those things in a later branch. Don't want to block it unnecessarily.

**Pull request merged**